docs/sources/helm-charts/mimir-distributed/migration-guides/migrate-enable-cluster-label-verification/_index.md

```diff
@@ -0,0 +1,96 @@
+---
+description: "Migrating Mimir Helm Installation to Use Memberlist Cluster Label"
+title: "Migrating Mimir Helm Installation to Use Memberlist Cluster Label"
+menuTitle: "Migrating Mimir Helm Installation to Use Memberlist Cluster Label"
+weight: 110
+---
+
+# Migrating Mimir Helm Installation to Use Memberlist Cluster Label
+
+## Introduction
+
+Grafana Mimir uses [memberlist] to share works and deciding which component replica to send the workload such as when ingesting series. Memberlist encodes the replica information in data structure called [hash ring], this is a consistent hashing scheme in which different ingesters instance tokens are placed around the ring. The token position in the ring determines which ingester should handle a request. The information of this hash ring is delivered to different components by using gossip protocol.
+
+By default, hash ring is global. If we also have Grafana Tempo and Grafana Loki in the same Kubernetes cluster, their components can unintentionally talk and sending data with each other, because Loki and Tempo also uses memberlist. This possibility to happen increases in cluster where the pods may reuse each other's IPs after churning.
```

Comment:
that first sentence is a bit unclear to me. I think saying that the key-value store where the hash ring is stored is viewed by all components.

re second sentence: I'd mention that loki and tempo clusters can end up participating in the same memberlist cluster and their key-value stores can merge, which in turn can lead to components from the different clusters to try to communicate with each other (I'd prefer "communicate" over "talk")

also re styling: I think "pod" should be "Pod" because that's the naming that k8s uses in its docs

Reply:
Most of this paragraph is removed. Let me know if you think we should keep information about the possible Pod reuse after churning.